GoMeet/src/main/webapp/js/create-meeting-step1.js

```diff
@@ -39,3 +39,48 @@ function addGuest(doc) {
  function removeGuest(guest, doc) {
   doc.getElementById(guest).remove();
 }
+
+/**
+ * Checks whether all the email input elements on the document are valid
+ * Note: Server side checks will also be made to validate emails 
+ * @param {Document} doc the document object containing the email inputs
+ */
+function validateEmails(doc) {
+  let allInputs = doc.getElementsByTagName('INPUT');
+
+  for (let i = 0; i < allInputs.length; i++) {
+    if (allInputs.item(i).type === 'email') {
+      if (allInputs.item(i).value === '') {
+        throw(new Error(BLANK_FIELDS_ALERT)); 
+      } else if (!isValidEmail(allInputs.item(i).value)) {
+        throw(new Error(INVALID_EMAILS_ALERT));
+      }
+    }
+  }
+  return true; 
+}
+
+/**
+ * Checks whether the given email address is valid according to the 
+ * regular expression pattern 
+ * @param {String} address the email address to be validated 
+ */
+function isValidEmail(address) {
```

Comment:
Hmm, ideally, the point of having `input` elements with `type=email` is that the browser should handle this for you. Check out [this](https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation)? The main thing to note is that they have their input fields within one `form` element, and also the button within the form element too, which should validate the inputs when pressed.

Reply:
I tried implementing it like this however combining the forms into one form broke my addGuest function. Also, I wanted to follow a similar structure to Anna's implementation of the submit button in create-meeting-step-2.html by calling a function that calls both the saveMeeting function and a validateXXX function.

Reply:
I think it'd be better to use something like (input.checkValidity())[https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement#methods] on the input element instead - this avoids the need for the complicated regex, and you can still use the same tests, but you have to pass in an input element to this function rather than a string.

Reply:
Done - thanks for the resource!

Reply:
Don't forget to update the tests

Reply:
Sorry that's my bad